addons/l10n_fr_invoice_addr/models/account_move.py

```diff
@@ -0,0 +1,41 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    def _get_default_xyz_partner(self, xyz):
+        partner = self.partner_id
+        if not partner:
+            return False
+
+        return self.env['res.partner'].browse(partner.address_get([xyz])[xyz])
+
+    @api.model
+    def _get_default_invoice_partner(self):
+        return self._get_default_xyz_partner('invoice')
+
+    partner_invoice_id = fields.Many2one('res.partner', readonly=True, tracking=True,
+        states={'draft': [('readonly', False)]},
```

Comment:
The goal was to prevent this field from being modified once the invoice leaves draft since this is the behavior of `partner_id` (and of a lot of fields in `account.move` really)

https://github.com/odoo/odoo/blob/b724fbcf6f348ed7a9109a325d366be69c052d1c/addons/account/models/account_move.py#L196-L199